src/fromager/resolver.py

```diff
@@ -243,3 +244,64 @@ def is_satisfied_by(self, requirement: Requirement, candidate: Candidate) -> boo
     def get_dependencies(self, candidate: Candidate) -> list:
         # return candidate.dependencies
         return []
+
+
+class GitHubTagProvider(ExtrasProvider):
+    def __init__(
+        self,
+        organization,
+        repo,
+    ):
+        super().__init__()
+        self.organization = organization
+        self.repo = repo
+        self.client = github.Github()
```

Comment:
GitHub has rate limiting for anon access based on IP addresses. Is there a way to pass credentials in case our infra runs into rate limiting?

Reply:
I've found tokens easy to use, so we could add that with a small environment variable check.

Reply:
#118